python_modules/dagster/dagster_tests/declarative_automation_tests/daemon_tests/test_e2e.py

```diff
@@ -836,3 +836,33 @@ def test_observable_source_asset_is_not_backfilled() -> None:
             assert len(runs) == 0
             backfills = _get_backfills_for_latest_ticks(context)
             assert len(backfills) == 0
+
+
+def test_dynamic_partitions() -> None:
```

Comment:
do you think it is worth testing the specific case where the lossy compression can in theory cause problems here? (removing a partition and then adding it back in, so the ordering of the keys gets shuffled)